src/datomish/query/clauses.cljc

```diff
@@ -62,6 +62,12 @@
                           (attribute-in-source (:source cc) value)
                           (constant-in-source (:source cc) value))]))
 
+(defn merge-ccs [left right]
+  (assoc left
+         :from (concat (:from left) (:from right))
```

Comment:
I somewhat expect these to be ordered sets rather than lists.  Just be certain (to yourself) that `concat` is appropriate here.